pkg/store/bucket.go

```diff
@@ -623,40 +642,33 @@ func newGCSBlock(
 		return nil, errors.Wrap(err, "read meta.json")
 	}
 
-	ix, err := newGCSIndex(ctx, dir, bkt.Object(path.Join(id.String(), "index")), gcsOperations)
+	ix, err := newBucketIndex(ctx, bkt, dir, path.Join(id.String(), "index"))
 	if err != nil {
 		return nil, errors.Wrap(err, "initialize index")
 	}
 
 	// Get object handles for all chunk files.
-	gcsOperations.WithLabelValues(gcsOperationObjectsList).Inc()
-
-	objs := bkt.Objects(ctx, &storage.Query{
-		Prefix: path.Join(id.String(), "chunks/"),
+	var chunkObjs []string
+	err = bkt.Iter(ctx, id.String()+"/chunks/", func(n string) error {
```

Comment:
`path.Join(id.String(), "chunks/")`

Reply:
Yea, `path.Join` kills the trailing slash, which then causes us to just load the `<id>/chunks/` object because we now always use the `Delimiter: "/"` query option behind the scenes.